app/lib/twitter_interface.rb

```diff
@@ -19,4 +24,44 @@ def config_client
       config.access_token_secret = Rails.application.secrets.access_secret
     end
   end
+
+  def check_twitter_activity board
+    return unless last_tweet = extract_info(get_last_tweet(board))
+    if board.last_tweet.present? && board.last_tweet.dig("url") != last_tweet["url"]
+      save_tweet_and_execute last_tweet, board
+    elsif Time.current - last_tweet["created_at"] < INTERVAL
+      save_tweet_and_execute last_tweet, board
```

Comment:
aici faci aceasi chestie pe ambele branchuri, poate fi combinata I guess dar cred ca se duce din lizibilitate. Up to you :))